go.mod

```diff
@@ -1,23 +1,23 @@
 module sigs.k8s.io/gateway-api
 
-go 1.21
+go 1.22.2
```

Comment:
there's no need to require  .2 (I think)

this is viral and will force all dependencies to do so as well

Reply:
I think it's automatic now, `go mod tidy` will hardcode `1.22.0` if we just have `1.22` here. Maybe that's preferable? (x-ref https://github.com/kubernetes/kubernetes/blob/master/go.mod#L9)

Reply:
Controversial...

https://twitter.com/liggitt/status/1780994018047840540
https://github.com/kubernetes/kubernetes/pull/124385

Reply:
It seems like upstream is moving to 1.22.2 for now (x-ref https://github.com/kubernetes/kubernetes/pull/124386), will stick with this until a better option emerges.

Reply:
I'd recommend `go 1.22.0` or whatever go dictates your minimum has to be ... including the patch means you won't fight go tooling, leaving it at the minimum patch in the minor means you are permissive to your downstreams and builders as possible

Reply:
Updated to use 1.22.0, thanks everyone!